torch_geometric/nn/aggr/lstm.py

```diff
@@ -0,0 +1,48 @@
+from typing import Optional
+
+from torch import Tensor
+from torch.nn import LSTM
+
+from torch_geometric.nn.aggr import Aggregation
+from torch_geometric.utils import to_dense_batch
+
+
+class LSTMAggregation(Aggregation):
+    r"""Performs LSTM-style aggregation in which the elements to aggregate are
+    interpreted as a sequence.
+
+    .. warn::
+        :class:`LSTMAggregation` is not permutation-invariant.
+
+    .. note::
+        :class:`LSTMAggregation` requires sorted indices.
+
+    Args:
+        in_channels (int): Size of each input sample.
+        out_channels (int): Size of each output sample.
+        **kwargs (optional): Additional arguments of :class:`torch.nn.LSTM`.
+    """
+    requires_sorted_index = True
+
+    def __init__(self, in_channels: int, out_channels: int, **kwargs):
+        super().__init__()
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.lstm = LSTM(in_channels, out_channels, batch_first=True, **kwargs)
+
+    def reset_parameters(self):
+        self.lstm.reset_parameters()
+
+    def forward(self, x: Tensor, index: Optional[Tensor] = None, *,
+                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
+                dim: int = -2) -> Tensor:
+
+        assert index is not None  # TODO
```

Comment:
sorry I didn't understand the reason for this, 
```
index = torch.zeros(x.size(0))
```
and 
```
index = None
```
will have the same result no?

Reply:
After working with the class not I see that this is because we don't yet support `ptr` - Should add a more informative exception here?